.travis.yml

```diff
@@ -18,9 +18,17 @@ before_install:
             sudo docker pull welder/web:latest
             sudo docker pull welder/web-with-coverage:latest
             sudo docker pull welder/web-e2e-tests:latest
-        else
+        fi
+install:
+  - |
+        if [ "$COMMAND" != "end-to-end-test" ]; then
             npm install
         fi
+        if [ "$COMMAND" == "cockpit-test" ]; then
+            node run build --with-coverage
+            sed -i "s,var welderApiHost=.*,var welderApiHost=\"api\";," ./public/js/config.js
+            sudo docker pull welder/web-e2e-tests:latest
```

Comment:
Move the `docker pull` in before_install section.

Reply:
Also `docker pull welder/web-cockpit:latest` because this is very unlikely to change often and `docker pull` inside Travis is faster than `dnf install`.